DatadogCore/Sources/Core/Upload/DataUploadWorker.swift

```diff
@@ -125,7 +125,7 @@ internal class DataUploadWorker: DataUploadWorkerType {
         queue.sync {
             while let nextBatch = self.fileReader.readNextBatch() {
                 _ = try? self.dataUploader.upload(events: nextBatch.events, context: contextProvider.read())
-                self.fileReader.markBatchAsRead(nextBatch)
+                self.fileReader.markBatchAsRead(nextBatch, reason: .flushed)
```

Comment:
blocking

we must not ignore the error with optional retry as this will hide the underlying error and we will always good flushed reason.

wrapping in a try-catch would be better option just like we are doing in `init`

Reply:
I added `try-catch` 👍 because it makes sense indeed. Now it tries once, and does one additional retry on failure.

However, this doesn't change the fact that we don't want to see `.flushed` batches in metrics - no matter if flushed successfully or not (this is why we don't attach the error information if "flushed" upload failed). This code is ran only in integration tests, cannot be reached from user-facing API and we don't want to skew our metric by this utility.

An alternative that I had consider is to make `reason` optional and pass `nil` in this flow. That could make it better reflect the requirement. However, dealing with optionality in downstream logic didn't seem safe-enough to me. It would invite future maintainers to basically skip the `reason` with passing `nil`, without considering what this option is used for and possibly skewing the metric. Defining certain set of "skipped" values looks IMO safer.

WDYT @ganeshnj ?

Reply:
Interesting, I didn't know this is test only code. I think I can be persuaded however we have another problem that we are writing code for testing which is shipped with the SDK instead we should design our APIs which are testable without additional code. 

Not related to this PR anyway, I'm good 👌